### DISCUSSION TITLE
Can I use Oracle 19c for development purpose? (Legal/License)

### DISCUSSION THREAD

**Discussion by bilalchraibi** [oracle/docker-images#1952] [Q&A] [answered]
In our development environment, we use the free flavors of Oracle Databases containers to develop and test our application. Here are the versions that we use:

- Oracle Database 11g Express Edition (XE)
- Oracle Database 18c Express Edition (XE)

One of our clients recently asked us if our software is compatible with Oracle Database 19c.
We want to give it a try in our development environment to see if the ODP.NET version that we use is compatible with it. But, we are not sure about the legal part of it.

So the question would be:

> Am I allowed to use it for Oracle Database 19c for development and testing purposes?

**Additional notes:**: 
- We are using the ODP.NET version 12 (4.121.2.0 ODAC RELEASE 4) in our software and we want to see if it is compatible with Oracle 19c.
- I found [this][2] Database license agreement that covers the development environment, but I'm not sure what it means when they use the word "***Customer***".

  [1]: https://github.com/oracle/docker-images/tree/main/OracleDatabase/SingleInstance#running-oracle-database-11gr2-express-edition-in-a-docker-container
  [2]: https://www.oracle.com/assets/databaselicensing-070584.pdf

**Comment by Djelibeybi** [comment, marked as answer]
> *Disclaimer*: I am **not** a lawyer and even if that were to ever change, I would still not be  **_your_** lawyer. 

Having said that, the license under which you download Oracle binaries is usually<sup>1</sup> determined by the location from which you download those binaries. For example, if you select an option on the [Oracle Database 19c](https://www.oracle.com/database/technologies/oracle-database-software-downloads.html#19c) page, you are required to agree to the terms of the [OTN Developer License](https://www.oracle.com/downloads/licenses/standard-license.html) which states:

_Oracle grants You a nonexclusive, nontransferable, limited license to internally use the Programs, subject to the restrictions stated in this Agreement, only for the purpose of developing, testing, prototyping, and demonstrating Your application and only as long as Your application has not been used for any data processing, business, commercial, or production purposes, and not for any other purpose._

It's at this point that I'm contractually obliged to recommend that you get legal counsel to review and confirm that your intended use of Oracle Database 19c downloaded from that page is allowed under the stated restrictions. Hopefully this at least provides you with the appropriate information if you chose to do so. :)

---

`1`: Customers with commercial support agreements should download binaries from <https://edelivery.oracle.com> so that the most appropriate license for their commercial agreement is presented during the download process.

**Reply by bilalchraibi** [reply]
Thank you @Djelibeybi for the clarifications.

**Comment by gvenzl** [comment]
Hi @bilalchraibi,

To add to what Avi already said, if you are curious about client/server compatibility you should also check out My Oracle Support notes:

* [Client / Server Interoperability Support Matrix for Different Oracle Versions (Doc ID 207303.1)](https://support.oracle.com/epmos/faces/DocContentDisplay?id=207303.1) 
* [Oracle Data Provider for .NET (ODP) Supported Configurations (Doc ID 726240.1)](https://support.oracle.com/epmos/faces/DocContentDisplay?id=726240.1)

The short answer is that the 12.2 client is compatible and at this point in time still fully supported with Oracle Database 19c.
However, I recommend upgrading to the 19c client at the earliest convenience. Oracle Database and Client 19c are both the LTS (long-term support) versions.

**Reply by bilalchraibi** [reply]
@gvenzl It is good to have a clear confirmation that it will work 😊. We have already added the client upgrade into our roadmap. We just want to be sure that the customers can still use our application even if they upgrade their oracle version.
